src/index.ts

```diff
@@ -23,9 +23,9 @@ const then = <T, U>(action: () => PromiseOrValue<T>, onFulfilled: (T) => Promise
 
 export interface PromiseOrValueModule {
   /** Apply a transformation to the result of a promise or value */
-  map: <T, U>(promiseOrValue: PromiseOrValue<T>, onFulfilled: (T) => PromiseOrValue<U>) => PromiseOrValue<U>
+  map: <T, U>(promiseOrValue: PromiseOrValue<T>, onFulfilled: (T: any) => PromiseOrValue<U>) => PromiseOrValue<U>
   /** Attach callbacks for the resolution/rejection of the result of the action */
-  then: <T, U>(action: () => PromiseOrValue<T>, onFulfilled: (T) => PromiseOrValue<U>, onRejected?: (reason: any) => PromiseOrValue<U>) => PromiseOrValue<U>
+  then: <T, U>(action: () => PromiseOrValue<T>, onFulfilled: (T: any) => PromiseOrValue<U>, onRejected?: (reason: any) => PromiseOrValue<U>) => PromiseOrValue<U>
```

Comment:
Here and the line above, the type should be `T` (from the generic type specified at the beginning - `<T, U>`) rather than the name of the argument being `T`.  You can write it as `(value: T)`. This should make the code safer as it will correctly limit the valid types. 

Think this might also be a bug in the current code - where it takes the `T` on its own and interprets that as the argument name instead of the argument type - so this PR will fix that too!